client/src/contexts/MealContext.js

```diff
@@ -0,0 +1,46 @@
+import React, { useReducer } from "react";
+import { useHistory, useLocation } from "react-router-dom";
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "createMeal":
+      return { meals: [...state.meals, action.payload.createdMeal], errorMessage: "" };
+    default:
+      return state;
+  }
+};
+
+const Context = React.createContext();
+
+const Provider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, {
+    meals: [],
+    errorMessage: "",
+  });
+
+  let history = useHistory();
+  let location = useLocation();
+
+
+  const createMeal = async({mealName, description}) => {
+    // do backend call
```

Comment:
the backend end point for creating a meal exists right ? is there a reason why you're not calling it ?